OKR.md

```diff
@@ -0,0 +1,50 @@
+# Quarterly Objectives and Key Results
+
+We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
+
+## 2018 Q4
+
+**go-ipfs handles large datasets (1TB++) without a sweat**
+- `PX` - It takes less than 48 hours to transfer 1TB dataset over Fast Ethernet (100Mbps)
+- `PX` - It takes less than 12 hours to transfer 200GB sharded dataset over Fast Ethernet (100Mbps)
+- `PX` - There is a prototype implementation of GraphSync for go-ipfs
+- `PX` - There is a better and more performant datastore module (e.g Badger or better)
+- `P1` - Rewrite pinning data structures to support large data sets / many files performantly
+
+**The bandwidth usage is reduced significantly and is well kept under control**
+- `PX` - Users can opt out of providing every IPLD node (and only provide root hashes)
+- `PX` - "Bitswap improvements around reducing chattiness, decreasing bandwidth usage (fewer dupe blocks), and increasing throughput"
+
+**It is a joy to use go-ipfs programatically**
+- `PX` @magik6k - The Core API is finalized and released. Make it easier to import go-ipfs as a package
+- `PX` - go-ipfs-api exposes the new Core API
```

Comment:
@diasdavid did you intend to let this also include the remote API protocol in this? Because I'm not sure it's covered in this document?

Reply:
My plan for this was to start with http api implementation and when the new rpc api becomes a thing support both as some users will likely need http is some setups

Reply:
Do you have a proposal for an updated KR that effectively captures that work? Taking PRs! =]

Reply:
@keks - Do you think the RPC API should be included (even partially) in this document? I don't know anything about that work but is it something that you think should be started in this next quarter?

Reply:
@magik6k is owner for this whole objective

making go-ipfs-api expose is lower priority (blocked by a lot of go-ipfs changes)

Reply:
To clarify, I was basically talking about [this issue](https://github.com/ipfs/go-ipfs-cmds/issues/37). We'll need to coordinate with the other implementors to find out what we really need, but the issue is that the current HTTP API has a lot of weird edge cases, which is painful for everyone who implements it.
Again, I'm not talking about the RPC methods we expose for others to call (these should mostly be the core api), but about the wire format used to make these calls. If we get that one right, a lot of stuff is going to be easier.